### PR TITLE
Features/improve api usdc support

### PR DESCRIPTION
**Pull request by Redm4x**
- Added usdc and usd stats to dashboard + graphs
- Update akt price of previous days
- Add missing `.next` cache to `.dockerignore` for reduced image sizes in Api & Indexer
- Upgraded `@cosmjs` to latest version in indexer to remove custom code. Could not update to latest in api because since [0.30.0](https://github.com/cosmos/cosmjs/blob/main/CHANGELOG.md#0300---2023-03-09) the `.toJSON()` method was removed. We will need to find an alternative for `.toJSON()` in the api.
- Improved typing in api and indexer

### CODE REVIEW

File: api/src/db/networkRevenueProvider.ts [hunk 1]
```diff
@@ -164,7 +179,7 @@ async function getDailyRevenue() {
     uakt: x.uakt,
     akt: uaktToAKT(x.uakt, 6),
     uusdc: x.uusdc,
-    usdc: udenomToDenom(x.uusdc),
+    usdc: udenomToDenom(x.uusdc, 6),
     usd: uaktToAKT(x.uakt, 6) * x.aktPrice + udenomToDenom(x.uusdc),
```
Comment (baktun14) [inline api/src/db/networkRevenueProvider.ts:183]: `udenomToDenom(x.uusdc, 6)`?

File: deploy-web/src/components/dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -128,30 +128,30 @@ export const Dashboard: React.FunctionComponent<IDashboardProps> = ({ dashboardD
...
-                  <FormattedNumber value={uaktToAKT(dashboardData.now.dailyUAktSpent)} maximumFractionDigits={2} />
+                  <FormattedNumber value={udenomToDenom(dashboardData.now.dailyUUsdSpent)} maximumFractionDigits={2} />
                 </span>
-                <AKTLabel />
+                <USDLabel />
               </>
             }
-            text="AKT spent (24h)"
+            text="USD spent (24h)"
             tooltip="Last 24h"
```
Comment (baktun14) [inline deploy-web/src/components/dashboard/Dashboard.tsx:137]: We could add a little description saying that it adds up the value in AKT + the USDC spent

File: deploy-web/src/components/dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -175,6 +175,76 @@ export const Dashboard: React.FunctionComponent<IDashboardProps> = ({ dashboardD
...
+        <Grid item xs={12} lg={3}>
+          <StatsCard
+            number={
+              <>
+                <FormattedNumber value={udenomToDenom(dashboardData.now.totalUUsdcSpent)} maximumFractionDigits={2} />
+                <USDCLabel />
+              </>
+            }
+            text="Total spent USDC"
+            tooltip="This is the total amount of usdc spent to rent computing power on the akash network since the beginning of the network. (March 2021)"
```
Comment (baktun14) [inline deploy-web/src/components/dashboard/Dashboard.tsx:241]: We could change this to the beginning of the USDC support

File: indexer/src/index.ts [hunk 4]
```diff
@@ -80,37 +81,38 @@ app.get("/nodes", async (req, res) => {
 });
 
 function startScheduler() {
-  scheduler.registerTask("Sync Blocks", syncBlocks, "7 seconds", true, {
-    id: env.HealthChecks_SyncBlocks
-  });
+  // scheduler.registerTask("Sync Blocks", syncBlocks, "7 seconds", true, {
```
Comment (baktun14) [inline indexer/src/index.ts:84]: Remove comment?

File: indexer/src/tasks/usdSpendingTracker.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
+import { AkashBlock } from "@shared/dbSchemas/akash";
+import { Day } from "@shared/dbSchemas/base";
+import { sequelize } from "@src/db/dbConnection";
+import { Op } from "sequelize";
+
+export async function updateUsdcSpending() {
```
Comment (baktun14) [inline indexer/src/tasks/usdSpendingTracker.ts:6]: `updateUsdSpending`?

File: api/src/db/networkRevenueProvider.ts (hunk 1 above)
Comment (Redm4x) [reply api/src/db/networkRevenueProvider.ts:183]: It's to match the precision of the akt amount

File: deploy-web/src/components/dashboard/Dashboard.tsx (hunk 2 above)
Comment (Redm4x) [reply deploy-web/src/components/dashboard/Dashboard.tsx:137]: Good idea 👍

File: indexer/src/index.ts (hunk 4 above)
Comment (Redm4x) [reply indexer/src/index.ts:84]: Oups, yeah I'll put back all scheduled tasks as before

**Review by baktun14 — APPROVED** [review]

**Pull request merged**